subsys/wifi/shell/shell.c

```diff
@@ -0,0 +1,612 @@
+				char *argv[])
+{
+	char *iface_name;
+
+	if (argc != 2 || !argv[1])
+		return -EINVAL;
+	iface_name = argv[1];
+
+	return wifimgr_ctrl_iface_open(iface_name);
+}
+
+static int wifimgr_cli_cmd_close(const struct shell *shell, size_t argc,
+				 char *argv[])
+{
+	char *iface_name;
+
+	if (argc != 2 || !argv[1])
+		return -EINVAL;
+	iface_name = argv[1];
+
+	return wifimgr_ctrl_iface_close(iface_name);
+}
+
+static int wifimgr_cli_cmd_scan(const struct shell *shell, size_t argc,
+				char *argv[])
+{
+	char *iface_name;
+
+	if (argc != 2 || !argv[1])
+		return -EINVAL;
+	iface_name = argv[1];
+
+	return wifimgr_ctrl_iface_scan(iface_name, wifimgr_cli_show_scan_res);
+}
+
+#ifdef CONFIG_WIFIMGR_STA
+static int wifimgr_cli_cmd_rtt_req(const struct shell *shell, size_t argc,
+				   char *argv[])
+{
+	struct wifi_rtt_request *rtt_req;
+	struct wifi_rtt_peers *peer;
+	int size;
+	int choice;
+	int ret;
+
+	size = sizeof(struct wifi_rtt_request) + sizeof(struct wifi_rtt_peers);
+	rtt_req = malloc(size);
+	if (!rtt_req)
+		return -ENOMEM;
+	memset(rtt_req, 0, size);
+	rtt_req->nr_peers = 1;
+	rtt_req->peers =
+	    (struct wifi_rtt_peers *)((char *)rtt_req +
+				      rtt_req->nr_peers *
+				      sizeof(struct wifi_rtt_request));
+	peer = rtt_req->peers;
+
+	optind = 0;
+	while ((choice = getopt(argc, argv, "b:c:m:")) != -1) {
+		switch (choice) {
+		case 'b':
+			peer->band = atoi(optarg);
+			if (!peer->band) {
+				printf("invalid band!\n");
+				return -EINVAL;
+			}
+			break;
+		case 'c':
+			peer->channel = atoi(optarg);
+			if (!peer->channel) {
+				printf("invalid channel!\n");
+				return -EINVAL;
+			}
+			break;
+		case 'm':
+			ret = strtomac(optarg, peer->bssid);
+			if (ret) {
+				printf("invalid BSSID!\n");
+				return ret;
+			}
+			break;
+		default:
+			return -EINVAL;
+		}
+	}
+
+	ret =
+	    wifimgr_ctrl_iface_rtt_request(rtt_req, wifimgr_cli_show_rtt_resp);
+	free(rtt_req);
+
+	return ret;
+}
+
+static int wifimgr_cli_cmd_connect(const struct shell *shell, size_t argc,
+				   char *argv[])
+{
+	return wifimgr_ctrl_iface_connect();
+}
+
+static int wifimgr_cli_cmd_disconnect(const struct shell *shell, size_t argc,
+				      char *argv[])
+{
+	return wifimgr_ctrl_iface_disconnect();
+}
+#endif
+
+#ifdef CONFIG_WIFIMGR_AP
+static int wifimgr_cli_cmd_start_ap(const struct shell *shell, size_t argc,
+				    char *argv[])
+{
+	return wifimgr_ctrl_iface_start_ap();
+}
+
+static int wifimgr_cli_cmd_stop_ap(const struct shell *shell, size_t argc,
+				   char *argv[])
+{
+	return wifimgr_ctrl_iface_stop_ap();
+}
+
+static int wifimgr_cli_cmd_del_sta(const struct shell *shell, size_t argc,
+				   char *argv[])
+{
+	char mac_addr[WIFI_MAC_ADDR_LEN];
+	int ret;
+
+	if (argc != 2 || !argv[1])
+		return -EINVAL;
+
+	ret = strtomac(argv[1], mac_addr);
+	if (ret) {
+		printf("invalid MAC address!\n");
+		return ret;
+	}
+
+	if (is_broadcast_ether_addr(mac_addr))
+		printf("Deauth all stations!\n");
```

Comment:
Please avoid using printf in command handlers. You have dedicated shell API for that:
Function: shell_fprintf or macros: `shell_print`, `shell_error`, `shell_info`, `shell_warn`.
Reason for that:
1. Shell is taking care to synchronize all messages on the screen. When you use printk you will compete for (usually) uart and displayed text can be in not correct order.
2. Shell is multiinstance so possibly you will not see this message at all. When you run shell on RTT and printf is connected to UART.

Reply:
@jarz-nordic 
Done, thanks!